src/main/java/edu/ie3/util/io/FileIOUtils.java

```diff
@@ -141,6 +150,320 @@ public static BufferedWriter getBufferedWriter(
+    } catch (IOException e) {
+      throw new FileException("Cannot create target directory '" + targetDirectory + "'.", e);
+    }
+
+    /* Monitor amount of entries and their size for safety reasons */
+    int entries = 0;
+    long size = 0;
+    try (InputStream fileInputStream = Files.newInputStream(archive);
+        BufferedInputStream bufferedInputStream = new BufferedInputStream(fileInputStream);
+        GzipCompressorInputStream gzipInputStream =
+            new GzipCompressorInputStream(bufferedInputStream);
+        TarArchiveInputStream tarInputStream = new TarArchiveInputStream(gzipInputStream)) {
+      ArchiveEntry archiveEntry;
+      while ((archiveEntry = tarInputStream.getNextEntry()) != null) {
+        /* Control the total amount of entries */
+        entries++;
+        if (entries > MAX_AMOUNT_OF_ENTRIES)
+          throw new IOException(
+              "The archive contains too many entries and is therefore possibly malicious.");
+
+        /* Control the size of extracted archive files */
+        long uncompressedSize = archiveEntry.getSize();
+        if (uncompressedSize == ArchiveEntry.SIZE_UNKNOWN)
+          throw new IOException(
+              "Unknown uncompressed file size of '" + archiveEntry.getName() + "'");
+        size += uncompressedSize;
+        if (size > MAX_SIZE_UNCOMPRESSED)
+          throw new IOException(
+              "Uncompressed size of archive exceeds permissible "
+                  + (MAX_SIZE_UNCOMPRESSED / 1024 / 1024)
+                  + " MB. Possibly malicious archive");
+
+        /* Control the compression ratio */
+        if (1 - (double) archiveSize / size > MAX_COMPRESSION_RATIO)
+          throw new IOException(
+              "Compression ratio exceeds its maximum permissible value "
+                  + (MAX_COMPRESSION_RATIO * 100)
+                  + " %. Possibly malicious archive");
+
+        handleZipEntrySafely(archiveEntry, targetDirectory, tarInputStream);
+      }
+    } catch (IOException ex) {
+      throw new FileException("Unable to extract from '" + archive + "'.", ex);
+    }
+
+    return targetDirectory;
+  }
+
+  /**
+   * Runs some pre-flight checks and assembles the target directory
+   *
+   * @param archive Compressed tarball archive to extract
+   * @param target Path to the target folder
+   * @return Path to the folder, where the content is meant to be extracted to
+   * @throws FileException If the pre-flight checks fail
+   */
+  private static Path determineTargetDirectory(Path archive, Path target) throws FileException {
+    /* Pre-flight checks */
+    if (Files.notExists(archive))
+      throw new FileException("There is no archive '" + archive + "' apparent.");
+    if (!Files.isRegularFile(archive))
+      throw new FileException("Archive '" + archive + "' is not a file.");
+    if (!archive.toString().endsWith(".tar.gz"))
+      throw new FileException("Archive '" + archive + "' does not end with '.tar.gz'.");
```

Comment:
Please use `TARGZ` instead of `".tar.gz"`

Reply:
Made the changes